src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryNullableDirective/CSharpRemoveRedundantNullableDirectiveDiagnosticAnalyzer.cs

```diff
@@ -47,6 +48,14 @@ protected override void InitializeWorker(AnalysisContext context)
                 context.RegisterSyntaxTreeAction(context =>
                 {
                     var root = context.Tree.GetCompilationUnitRoot(context.CancellationToken);
+                    if (context.FilterSpan.HasValue)
+                    {
+                        // Bail out if the analysis filter span does not have any nullable directives.
+                        var node = root.FindNode(context.FilterSpan.GetValueOrDefault(), findInsideTrivia: true, getInnermostNodeForTie: true);
+                        if (!node.DescendantNodesAndSelf(descendIntoTrivia: true).Any(node => node is NullableDirectiveTriviaSyntax))
+                            return;
```

Comment:
it would be good if you first even checked if hte node.HasDirectives before bothering to walk into it.

Reply:
Switched to calling the `ContainsDirective(SyntaxKind)` API, which has this check upfront.